web/src/app/shared/components/upgrade.component.ts

```diff
@@ -71,16 +82,21 @@ export class UpgradeComponent extends BaseDirective {
   @Input()
   set row(row: UpgradeItem) {
     this.pRow = row;
-    if (row.upgrade) {
-      this.list$ = this.api.get(`${row.upgrade}?ordering=-name`).pipe(filter((list: Upgrade[]) => !!list.length));
-    }
   }
 
   @Output()
   refresh: EventEmitter<EmmitRow> = new EventEmitter<EmmitRow>();
 
   constructor(private api: ApiService, private dialog: MatDialog) {
     super();
+
+    this._list$ = new BehaviorSubject<Upgrade[]>([]);
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
```

Comment:
We need to find out another way to subscribe to all the application changes

Reply:
What is wrong with this hook?

Reply:
https://angular.io/api/core/OnChanges
"OnChanges

A lifecycle hook that is called when any data-bound property of a directive changes. Define an ngOnChanges() method to handle the changes."

So I think this place is better than a setter.

Reply:
What the changes between old setter and the hook? Why did you choose the hook?

Reply:
Because this is a special place for this kind of action.

Reply:
OnChanges - a lifecycle hook that is called when any data-bound property of a directive changes. In OnChanges hook we have a check when any data-bound property of a directive changes. A setter helps us to avoid the check.